### PR TITLE
Prefix form id when it is present in form_for/3

### PR DESCRIPTION
**Pull request by feliperenan**
It avoid issues when having multiple forms on the same page, you can see
more info about it here:

phoenixframework/phoenix_live_view#1014

### Phoenix HTML update

In order to test `form_for/3` here I had to update phoenix html. Please, let me know if that's ok 🙌

### CODE REVIEW

File: mix.lock [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+  "makeup_elixir": {:hex, :makeup_elixir, "0.14.0", "cf8b7c66ad1cff4c14679698d532f0b5d45a3968ffbcbfd590339cb57742f1ae", [:mix], [{:makeup, "~> 1.0", [hex: :makeup, repo: "hexpm", optional: false]}], "hexpm", "d4b316c7222a85bbaa2fd7c6e90e37e953257ad196dc229505137c5e505e9eff"},
+  "mime": {:hex, :mime, "1.3.1", "30ce04ab3175b6ad0bdce0035cba77bba68b813d523d1aac73d9781b4d193cf8", [:mix], [], "hexpm", "6cbe761d6a0ca5a31a0931bf4c63204bceb64538e664a8ecf784a9a6f3b875f1"},
+  "nimble_parsec": {:hex, :nimble_parsec, "0.5.1", "c90796ecee0289dbb5ad16d3ad06f957b0cd1199769641c961cfe0b97db190e0", [:mix], [], "hexpm", "00e3ebdc821fb3a36957320d49e8f4bfa310d73ea31c90e5f925dc75e030da8f"},
+  "phoenix_html": {:hex, :phoenix_html, "2.14.2", "b8a3899a72050f3f48a36430da507dd99caf0ac2d06c77529b1646964f3d563e", [:mix], [{:plug, "~> 1.5", [hex: :plug, repo: "hexpm", optional: false]}], "hexpm", "58061c8dfd25da5df1ea0ca47c972f161beb6c875cd293917045b92ffe1bf617"},
```
Comment (feliperenan) [inline mix.lock:12]: @josevalim my goal was just to update Phoenix HTML. I'm not sure why I'm getting all those updates though 🙌

File: test/phoenix_ecto/html_test.exs [hunk 2]
```diff
@@ -204,10 +254,6 @@ defmodule PhoenixEcto.HTMLTest do
       assert input_value(f, :integer) == 123
       assert input_value(f, :string) == "string"
       assert input_value(f, :float) == 78.9
-
-      assert input_value(f, :integer, 0) == 123
-      assert input_value(f, :string, "default") == "default"
-      assert input_value(f, :float, 0.0) == 78.9
```
Comment (feliperenan) [inline test/phoenix_ecto/html_test.exs:210]: This has been removed because we don't have `input_value/3` on Phoenix HTML anymore.

**Comment by feliperenan** [comment]
build is breaking because Phoenix HTML update also updates plug which requires Elixir 1.7. Probably bumping Elixir here would fix the issue as well. I'm not sure if that's fine though 🙌

File: lib/phoenix_ecto/html.ex [hunk 3]
```diff
@@ -3,12 +3,15 @@ if Code.ensure_loaded?(Phoenix.HTML) do
     def to_form(changeset, opts) do
       %{params: params, data: data} = changeset
       {name, opts} = Keyword.pop(opts, :as)
+      {id, opts} = Keyword.pop(opts, :id)
+
       name = to_string(name || form_for_name(data))
+      id = if is_nil(id), do: name, else: to_string(id) <> "_#{name}"
```
Comment (josevalim) [inline lib/phoenix_ecto/html.ex:9]: I am thinking either `id || name`, without prefixing the `_#{name}` if an ID is given. WDYT?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**